e2e/internal/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,378 @@
+			if pod.Status.Phase != coreV1.PodRunning {
+				fmt.Printf("Pod %s is not running, but %s\n", pod.Name, pod.Status.Phase)
+				return false
+			}
+
+			for _, condition := range pod.Status.Conditions {
+				switch condition.Type {
+				case coreV1.ContainersReady:
+					if condition.Status != coreV1.ConditionTrue {
+						fmt.Printf("Pod %s's status %s is false\n", pod.Name, coreV1.ContainersReady)
+						return false
+					}
+				case coreV1.PodInitialized:
+					if condition.Status != coreV1.ConditionTrue {
+						fmt.Printf("Pod %s's status %s is false\n", pod.Name, coreV1.PodInitialized)
+						return false
+					}
+				case coreV1.PodReady:
+					if condition.Status != coreV1.ConditionTrue {
+						fmt.Printf("Pod %s's status %s is false\n", pod.Name, coreV1.PodReady)
+						return false
+					}
+				case coreV1.PodScheduled:
+					if condition.Status != coreV1.ConditionTrue {
+						fmt.Printf("Pod %s's status %s is false\n", pod.Name, coreV1.PodScheduled)
+						return false
+					}
+				}
+			}
+		}
+		return true
+	}, time.Minute*30, time.Minute)
+}
+
+// testNodesEventuallyAvailable checks that:
+// 1) all nodes only have good status conditions.
+// 2) the expected number of nodes have joined the cluster.
+func testNodesEventuallyAvailable(t *testing.T, k *kubernetes.Clientset, wantControlNodeCount, wantWorkerNodeCount int) {
+	require.Eventually(t, func() bool {
+		nodes, err := k.CoreV1().Nodes().List(context.Background(), metaV1.ListOptions{})
+		if err != nil {
+			fmt.Println(err)
+			return false
+		}
+
+		var controlNodeCount, workerNodeCount int
+		for _, node := range nodes.Items {
+			if _, ok := node.Labels["node-role.kubernetes.io/control-plane"]; ok {
+				controlNodeCount++
+			} else {
+				workerNodeCount++
+			}
+
+			for _, condition := range node.Status.Conditions {
+				switch condition.Type {
+				case coreV1.NodeReady:
+					if condition.Status != coreV1.ConditionTrue {
+						fmt.Printf("Status %s for node %s is %s\n", coreV1.NodeReady, node.Name, condition.Status)
+						return false
+					}
+				case coreV1.NodeMemoryPressure:
+					if condition.Status != coreV1.ConditionFalse {
+						fmt.Printf("Status %s for node %s is %s\n", coreV1.NodeMemoryPressure, node.Name, condition.Status)
+						return false
+					}
+				case coreV1.NodeDiskPressure:
+					if condition.Status != coreV1.ConditionFalse {
+						fmt.Printf("Status %s for node %s is %s\n", coreV1.NodeDiskPressure, node.Name, condition.Status)
+						return false
+					}
+				case coreV1.NodePIDPressure:
+					if condition.Status != coreV1.ConditionFalse {
+						fmt.Printf("Status %s for node %s is %s\n", coreV1.NodePIDPressure, node.Name, condition.Status)
+						return false
+					}
+				case coreV1.NodeNetworkUnavailable:
+					if condition.Status != coreV1.ConditionFalse {
+						fmt.Printf("Status %s for node %s is %s\n", coreV1.NodeNetworkUnavailable, node.Name, condition.Status)
+						return false
+					}
+				}
+			}
```

Comment:
```suggestion
				switch condition.Type {
				case coreV1.NodeReady, coreV1.NodeMemoryPressure, coreV1.NodeDiskPressure, coreV1.NodePIDPressure, coreV1.NodeNetworkUnavailable:
					if condition.Status != coreV1.ConditionTrue {
						fmt.Printf("Status %s for node %s is %s\n", condition.Type, node.Name, condition.Status)
						return false
					}
				}
			}
```
Not 100% sure I understand this code segment correctly, but this seems a lot simpler?